lib/ssl/doc/src/using_ssl.xml

```diff
@@ -119,74 +126,77 @@ ok</code>
-      otherwise TLS handshake messages can be delivered to the wrong
-      process:</p>
 
-      <code type="erl">4 server> inet:setopts(Socket, [{active, false}]).
-ok</code>
+  <section>
+  <title>Upgrade Example - TLS only</title>
+
+  <p>Upgrading a a TCP/IP connection to a TLS connections is mostly
+  used when there is a desire have unencrypted communication first and
+  then later secure the communication channel by using TLS. Note that
+  the client and server need to agree to do the upgrade in the
+  protocol doing the communication. This is concept is often
+  referenced as <c>STARTLS</c> and used in many protocols such as
+  <c>SMTP</c>, <c>FTPS</c> and <c>HTTPS</c> via a proxy.
+  </p>
+
+  <warning><p>Maximum security recommendations are however moving away from such solutions.</p></warning>
       
-      <p><em>Step 6:</em> Do the TLS handshake:</p>
-      <code type="erl">5 server> {ok, TLSSocket} = ssl:handshake(Socket, [{cacertfile, "cacerts.pem"},
-{certfile, "cert.pem"}, {keyfile, "key.pem"}]).
-{ok,{sslsocket,[...]}}</code>
+  <p>To upgrade to a TLS connection:</p>
+
+  <p><em>Step 1:</em> Start the server side:</p>
+  <code type="erl">1 server> ssl:start().
+  ok</code>
+
+  <p><em>Step 2:</em> Create a normal TCP listen socket and ensure
+  <c>active</c> is set to <c>false</c> and not set to any active mode
+  otherwise TLS handshake messages can be delivered to the
+  wrong process.
+  </p>
+  <code type="erl">2 server> {ok, ListenSocket} = gen_tcp:listen(9999, [{reuseaddr, true},
+  {active, false}]).
+  {ok, #Port&lt;0.475&gt;}</code>
+
+  <p><em>Step 3:</em> Accept client connection:</p>
+  <code type="erl">3 server> {ok, Socket} = gen_tcp:accept(ListenSocket).
+  {ok, #Port&lt;0.476&gt;}</code>
+
+  <p><em>Step 4:</em> Start the client side:</p>
+  <code type="erl">1 client> ssl:start().
+  ok</code>
+
+  <code type="erl">2 client> {ok, Socket} = gen_tcp:connect("localhost", 9999,  [], infinity).</code>
+
+  <p><em>Step 5:</em> Do the TLS handshake:</p>
+  <code type="erl">4 server> {ok, TLSSocket} = ssl:handshake(Socket, [{verify, verify_peer},
+  {cacertfile, "cacerts.pem"},
+  {certs_keys, [#{certfile => "cert.pem", keyfile => "key.pem"}]}]).
+  {ok,{sslsocket,[...]}}</code>
       
-      <p><em>Step 7:</em> Upgrade to a TLS connection. The client and
-      server must agree upon the upgrade. The server must call
-      <c>ssl:handshake/2</c> before the client calls
-      <c>ssl:connect/3.</c></p>
+  <p><em>Step 6:</em> Upgrade to a TLS connection. The client and
+  server must agree upon the upgrade. The server must be prepared to be a TLS server before the client
+  can do a successful connect.</p>
 
-      <code type="erl">3 client>{ok, TLSSocket} = ssl:connect(Socket, [{cacertfile, "cacerts.pem"},
-{certfile, "cert.pem"}, {keyfile, "key.pem"}], infinity).
+  <code type="erl">3 client>{ok, TLSSocket} = ssl:connect(Socket, [{cacertfile, "cacerts.pem"},
```

Comment:
here and in other places:
why specyfing `cacertfile` without `verify_peer` ?

Is `cacertfile` used without `verify_peer`

Reply:
Added the verify_peer option, I think we should. Also took the opportunity to add the server option fail_if_no_peer_cert to the example. However, cacerts are also used to build the own chain if it is not explicitly provided.